File: codewars/ArrayDiff.php
function arrayDiff($a, $b) {
    $result = array();
    
    foreach ($a as $value) {
        if (!in_array($value, $b)) {
            $result[] = $value;
        }
    }
    
    return $result;
}

$a = array(1, 2, 2, 2, 3);
$b = array(2);

$z = arrayDiff($a, $b);

echo "Resultado: ";
print_r($z);

File: codewars/ArrayDiff.cpp
#include <iostream>
#include <vector>
#include <algorithm>

std::vector<int> array_diff(const std::vector<int>& a, const std::vector<int>& b) {
    std::vector<int> result;
    
    for (const int& value : a) {
        if (std::find(b.begin(), b.end(), value) == b.end()) {
            result.push_back(value);
        }
    }
    
    return result;
}

int main() {
    std::vector<int> a = {1, 2, 2, 2, 3};
    std::vector<int> b = {2};
    
    std::vector<int> z = array_diff(a, b);

    std::cout << "Resultado: ";
    for (const int& value : z) {
        std::cout << value << " ";
    }
    std::cout << std::endl;

    return 0;
}
